### PR TITLE
[PUB-2571] Small improvements in posts

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description
- Create postContent in postParser, to group data needed for UpdateContent.
- Move RetweetPanel inside UpdateContent
- Simplify PostComponent logic in QueueItems and PostList
- Re-add video tag to video updates

## Context & Notes
I've decided to add the postContent instead of replacing several properties in the postParser, as a proper parser cleaning requires mores time and have in consideration the data consumed by the composer
https://buffer.atlassian.net/browse/PUB-2571

## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/server/parsers/src/postParser.js [hunk 1]
```diff
@@ -221,5 +221,21 @@ module.exports = post => {
...
+        description: media.description,
+        thumbnailUrl: media.preview,
+      },
+      links: canHaveLinks ? links : [],
+      retweetComment,
+      retweetCommentLinks,
+      retweetProfile: getRetweetProfileInfo(post),
+      text,
+      type: getPostType({ post }),
+    },
```
Comment (karelm) [inline packages/server/parsers/src/postParser.js:239]: this is so much nicer, having a well defined structure for a post 😍

File: packages/shared-components/PostList/index.jsx [hunk 2]
```diff
@@ -30,15 +30,6 @@ const HeaderStyle = styled.div`
...
 `;
 
-const postTypeComponentMap = new Map([
-  ['text', Post],
-  ['image', Post],
-  ['multipleImage', Post],
-  ['link', Post],
-  ['video', Post],
-  ['storyGroup', Story],
-]);
```
Comment (karelm) [inline packages/shared-components/PostList/index.jsx:40]: ❤️

File: packages/shared-components/UpdateContent/index.jsx [hunk 3]
```diff
@@ -4,53 +4,54 @@ import styled from 'styled-components';
...
-    </ContentWrapper>
+    <React.Fragment>
+      {retweetProfile ? (
+        <RetweetPanel {...props}>
+          <Content {...props} isLink={isLink} isMedia={isMedia} />
+        </RetweetPanel>
+      ) : (
+        <Content {...props} isLink={isLink} isMedia={isMedia} />
+      )}
+    </React.Fragment>
```
Comment (karelm) [inline packages/shared-components/UpdateContent/index.jsx:39]: This.... just this 🙌 😍  WOW

File: packages/shared-components/UpdateMediaContent/index.jsx [hunk 4]
```diff
@@ -83,9 +84,9 @@ const UpdateMediaContent = ({ imageSrc, imageUrls, tag, type }) => {
         backdropClosesModal
         showImageCount={false}
       />
-      {isSingleImage && tag && (
+      {isVideo && (
         <ThumbnailTag>
-          <IdTag>{tag}</IdTag>
+          <IdTag>VIDEO</IdTag>
         </ThumbnailTag>
       )}
```
Comment (karelm) [inline packages/shared-components/UpdateMediaContent/index.jsx:91]: do we not show tags for images then now? or is tag only set for "Video"? if the latter, I completely get this change 👍

Comment (anafilipadealmeida) [reply packages/shared-components/UpdateMediaContent/index.jsx:91]: What I realized is that this prop tag was coming from nowhere. The VideoPost and VideoDraft had it as a defaultProp: VIDEO 🤷‍♀️ Which means it was only used in Video

**Pull request merged**